compiler/rustc_hir_analysis/src/hir_ty_lowering/dyn_compatibility.rs

```diff
@@ -70,13 +69,6 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             let guar = self.report_trait_object_addition_traits(&regular_traits);
             return Ty::new_error(tcx, guar);
         }
-        // We don't support `PointeeSized` principals
```

Comment:
My PR predates #143104.

This commit basically reverts this logic because it's unreachable due to us filtering out `PointeeSized` inside `lower_poly_trait_ref`. The diagnostic is definitely worse, we now say `at least one trait is required for an object type` instead of `` `PointeeSized` cannot be used with trait objects ``.

I guess I should address the diagnostic regression since it only makes sense if you know the lowering. This ties in with https://github.com/rust-lang/rust/pull/142693#discussion_r2173559498.